src/Oracle/Protocol/OracleHttpProtocol.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Data;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using JArray = Newtonsoft.Json.Linq.JArray;
+using JObject = Newtonsoft.Json.Linq.JObject;
+
+namespace Neo.Plugins
+{
+    public class OracleHttpProtocol : IOracleProtocol
+    {
+        public int Timeout { get; set; } = 5000;
+        public bool AllowPrivateHost { get; set; } = false;
+        public readonly string[] AllowedFormats = new string[] { "application/json" };
+
+        public byte[] Request(ulong requestId, string url, string filter)
+        {
+            Utility.Log(nameof(OracleHttpProtocol), LogLevel.Debug, $"Request: {url}");
+
+            if (!Uri.TryCreate(url, UriKind.Absolute, out var uri)) throw new InvalidOperationException("UrlError");
+            if (!AllowPrivateHost && IsInternal(Dns.GetHostEntry(uri.Host))) throw new InvalidOperationException("Access to private host is not allowed");
+
+            using var handler = new HttpClientHandler();
+            using var client = new HttpClient(handler);
+            client.DefaultRequestHeaders.Add("Accept", string.Join(",", AllowedFormats));
+
+            Task<HttpResponseMessage> result = client.GetAsync(uri);
+            Stopwatch sw = new Stopwatch();
```

Comment:
We can improve this code in a future PR, but there are better ways to handle HttpClient timeouts: https://thomaslevesque.com/2018/02/25/better-timeout-handling-with-httpclient/

Reply:
As we also need to consider the parsing time, then we need to record the time consumed by the request.